### PR TITLE
tests: add gsettings interface spread test

### PR DESCRIPTION
**Pull request by fgimenez**

### CODE REVIEW

File: tests/main/interfaces-gsettings/task.yaml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    snap interfaces | grep -Pzq "$CONNECTED_PATTERN"
+
+    echo "==================================="
+
+    echo "When the plug is connected"
+    snap connect gsettings-consumer:gsettings ubuntu-core:gsettings
+    snap interfaces | grep -Pzq "$CONNECTED_PATTERN"
+
+    echo "Then the snap is able to write the gsettings user configuration"
+    sudo -i -u test /bin/sh -c 'gsettings-consumer.write /home/test/.config/dconf/user gsettings_content'
```
Comment (niemeyer) [inline tests/main/interfaces-gsettings/task.yaml]: This doesn't feel like a very realistic gsettings client. It's a lot of logic just to ensure we have these two path lines on the apparmor profile, and would blindly continue passing even if everything else is broken and gsettings doesn't actually work.

Comment (fgimenez) [inline tests/main/interfaces-gsettings/task.yaml]: @niemeyer indeed very good catch, I've prepared a test snap with a gsettings client, will wait for it to be uploaded to the store and push the changes to use it here.

Thanks,

**Comment by fgimenez** [comment]
@niemeyer done, the test is throwing an error `No schemas installed` but no apparmor related problems in dmesg, running with `sudo strace -f /snap/bin/test-snapd-gsettings-consumer.get`:
```
[pid 14815] open("/usr/local/share/glib-2.0/schemas/gschemas.compiled", O_RDONLY) = -1 ENOENT (No such file or directory)
[pid 14815] futex(0x7f3691d99428, FUTEX_WAKE_PRIVATE, 2147483647) = 0
[pid 14815] write(2, "No schemas installed\n", 21No schemas installed
```

@jdstrand could you please have a look?

Thanks!

**Comment by zyga** [comment]
Can we please merge master into this to see what is the current status?

**Comment by fgimenez** [comment]
@zyga done thx for having a look :)

**Comment by mvo5** [comment]
Tests fail: https://travis-ci.org/snapcore/snapd/jobs/159398750#L493

**Comment by niemeyer** [comment]
Tests still fail, and related to the new test. This has been up for more than a month without it working. @fgimenez, please reopen the PR when there's something we can merge.

**Pull request closed without merging**